src/data/IFormState.ts

```diff
@@ -0,0 +1,24 @@
+// The FormStatus is to represent FormInput status
+export interface IFormState {
+    [id: string]: IFieldState;
+}
+
+export interface IFieldState {
+    error: string,
+}
+
+export interface INestedFieldState extends IFieldState {
+    nestedStatus: Array<IFormState>;
```

Comment:
Should this be `nestedStatus: IFormState`? Since IFormState is a map of fields to errors, I don't understand why Array is necessary